games/src/main/java/brocoeur/example/games/controller/GamesController.java

```diff
@@ -56,11 +56,23 @@ private void processDirectMsg(final ServiceRequest serviceRequest) {
         final GameStrategyTypes gameStrategyTypes = serviceRequest.getGameStrategyTypes();
         final GameStrategy gameStrategy = gameStrategyTypes.getGameStrategy();
 
-        final GamePlay gamePlayFromUser = gameStrategy.getStrategyPlay();
+        GamePlay gamePlayFromUser = gameStrategy.getStrategy();
+
         final GamePlay gamePlayFromService = gameService.play(gameStrategyTypes.getGameTypes());
-        LOGGER.info("[DIRECT] - USER plays: '" + gamePlayFromUser + "' and SERVICE plays: '" + gamePlayFromService + "'.");
 
-        if (gamePlayFromUser.equals(gamePlayFromService)) {
+        //all the cases which are about user giving strategies and not value to compare ↓
+        switch (gameStrategyTypes.getGameTypes()){
+            case BLACK_JACK :
+                gamePlayFromUser = gameService.play(gameStrategyTypes.getGameTypes(),gameStrategy);
+
+
+            //more cases to come in the future games
+
+        }
+
+        LOGGER.info("[DIRECT] - USER's play: '" + gamePlayFromUser + "' and SERVICE' play: '" + gamePlayFromService + "'.");
+
+        if (gameService.didPlayerWin(gameStrategyTypes.getGameTypes(),gamePlayFromUser,gamePlayFromService)) {
             LOGGER.info("User WON !");
```

Comment:
All this should be in Offline.

https://github.com/gaugerard/brocoeur/blob/bf0eaba3649c9f1e968356f367abc5d051a0bb9d/games/src/main/java/brocoeur/example/games/controller/GamesController.java#L85

here we can use the previous game result in the blackack strategy. This is what you want to do right ?

Reply:
No this is not what I want to do